### PR TITLE
fix(clerk-js): Minor UI tweaks

### PR DESCRIPTION
**Pull request by octoper**
## Description

This PR include a lot of minor UI tweaks for the re-theme of the components:
- `<SIgnUp/>` ,`<SignIn/>` lot's of improvements to match the designs and look better on mobile devices
- Phone input improvements
- Input error improvements
- Added `danger` variant for Alert component
- Fixed keyboard navigation for when picking members role

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by octoper** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @octoper, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-8zuwp9nfb.clerkstage.dev) | Dec 22, 2023 10:30 PM |

File: packages/clerk-js/src/ui/elements/Header.tsx [hunk 1]
```diff
@@ -34,7 +34,12 @@ const Subtitle = React.memo((props: PropsOfComponent<typeof Text>): JSX.Element
       elementDescriptor={descriptors.headerSubtitle}
       variant='body'
       colorScheme='neutral'
-      sx={sx}
+      sx={[
+        t => ({
+          lineHeight: t.lineHeights.$normal,
```
Comment (anagstef) [inline packages/clerk-js/src/ui/elements/Header.tsx:39]: 👀

Comment (octoper) [reply packages/clerk-js/src/ui/elements/Header.tsx:39]: Good catch, fixing!

File: packages/clerk-js/src/ui/elements/Pagination.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ export type PageButtonProps = PropsOfComponent<typeof Button> & {
 };
 
 const PageButton = (props: PageButtonProps) => {
-  const { sx, isActive, children, ...rest } = props;
+  const { sx, isActive, icon, children, ...rest } = props;
```
Comment (anagstef) [inline packages/clerk-js/src/ui/elements/Pagination.tsx:30]: 💯

**Review by anagstef — APPROVED** [review]

**Pull request merged**